modules/tfm/tfm/boards/board/device_cfg.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#ifndef DEVICE_CFG_H__
+#define DEVICE_CFG_H__
+
+/**
+ * \file device_cfg.h
+ * \brief
+ * This is the default device configuration file with all peripherals
+ * defined and configured to be use via the secure and/or non-secure base
+ * address.
+ */
+
+#undef ARRAY_SIZE
```

Comment:
What is this for?

Reply:
aliasing issue with zephyr's util.h

util.h already has prior-art for ifndef guarding MAX and MIN so an ifndef guard of ARRAY_SIZE should go through easily upstream.

Please post a PR to this effect to upstream Zephyr.

Use ARRAY_SIZE being as common as MAX and MIN and the prior ifndef guard's in the rationale of the PR.

Reply:
```
diff --git a/include/zephyr/sys/util.h b/include/zephyr/sys/util.h
index fdc4532cf8..d2ed9a26b8 100644
--- a/include/zephyr/sys/util.h
+++ b/include/zephyr/sys/util.h
@@ -96,6 +96,7 @@ extern "C" {
                !__builtin_types_compatible_p(__typeof__(array), \
                                              __typeof__(&(array)[0])))
 
+#ifndef ARRAY_SIZE
 /**
  * @brief Number of elements in the given @p array
  *
@@ -107,7 +108,7 @@ extern "C" {
  */
 #define ARRAY_SIZE(array) \
        ((size_t) (IS_ARRAY(array) + (sizeof(array) / sizeof((array)[0]))))
-
+#endif
 #endif /* __cplusplus */
 
 /**

```

Reply:
Created
https://github.com/zephyrproject-rtos/zephyr/pull/50855

Reply:
As pointed out by carles there is already a PR upstream for this change: https://github.com/zephyrproject-rtos/zephyr/pull/50239